code/game/objects/items/devices/radio/radio.dm

```diff
@@ -429,16 +431,14 @@
 		return
 
 	var/mob/living/holder = loc
-	if(!holder.client?.prefs.read_preference(/datum/preference/toggle/radio_noise) && !HAS_TRAIT(holder, TRAIT_DEAF))
-		return
-
-	var/list/spans = data["spans"]
-	if(COOLDOWN_FINISHED(src, audio_cooldown))
-		COOLDOWN_START(src, audio_cooldown, 0.5 SECONDS)
-		SEND_SOUND(holder, 'sound/misc/radio_receive.ogg')
-	if((SPAN_COMMAND in spans) && COOLDOWN_FINISHED(src, important_audio_cooldown))
-		COOLDOWN_START(src, important_audio_cooldown, 0.5 SECONDS)
-		SEND_SOUND(holder, 'sound/misc/radio_important.ogg')
+	if(holder.client?.prefs.read_preference(/datum/preference/toggle/radio_noise) && !HAS_TRAIT(holder, TRAIT_DEAF) && radio_noise)
```

Comment:
Readd the early return.

Reply:
But if I re-add the early return - the code of radio/entertainment/speakers won't work anymore as it works if this code continues.

What I should do?

Reply:
whatever I got it